src/api/services/dotcms-events-service.ts

```diff
@@ -24,16 +24,12 @@ export class DotcmsEventsService {
      * @param dotcmsConfig - The dotCMS configuration properties that include
      *                        the Websocket parameters.
      */
-    constructor(@Inject('dotcmsConfig') private dotcmsConfig: DotcmsConfig, private loginService: LoginService) {
+    constructor(private dotcmsConfig: DotcmsConfig, private loginService: LoginService) {
         this.protocol = dotcmsConfig.getWebsocketProtocol();
         this.baseUrl = dotcmsConfig.getWebsocketBaseUrl();
         this.endPoint = dotcmsConfig.getSystemEventsEndpoint();
 
-        if (loginService.loginUser) {
-            this.connectWithSocket(this.loginService.loginUser);
-        }
-
-        this.loginService.loginUser$.subscribe(user => this.connectWithSocket(user));
+        loginService.watchUser(this.connectWithSocket.bind(this));
```

Comment:
Meh... I like it better this way, because we are explicitly passing the context where to execute that method. 

So 50-50... any other opinion people?